base_localization:
  ros__parameters:

    # tópicos
    detection_topic: "/base_detection/detected_coords"
    num_bases_topic: "/base_detection/num_bases"

    # frames e TF
    reference_frame: "map"
    target_frame: "base_link_1"
    tf_timeout_sec: 0.2

    # QoS das detecções
    detection_qos_reliable: true

    # centro da imagem (px)
    cx_ref: 480.0
    cy_ref: 360.0

    # markers
    marker_topic: "/base_localization/line"
    marker_ns: "base_link_lines"
    marker_line_width: 0.03
    marker_color_r: 0.1
    marker_color_g: 0.9
    marker_color_b: 0.1
    marker_color_a: 1.0
    marker_lifetime_sec: 0.0

    # saída
    poses_topic: "/base_localization/poses"

    # --- poly_size ---
    poly_size.D0: 300.0

    # ax (X): [a_00, a_01, a_02,   a_10, a_11, a_12,   a_20, a_21, a_22]
    poly_size.ax: [0.0777577409545744, 0.000096750545203726, -0.00010449891819678,
                   0.0058051857140838, -0.00196248970695934, 0.0011451591888037,
                   0.000105124036551204, 0.000544136722041704, 0.0007034535632977]

    # ay (Y): mesma estrutura
    poly_size.ay: [0.0822471187270925, -0.000341372798534558, 0.0000763129652099313,
                   0.00399228719434308, -0.0061448685885743, 0.00279547058545915,
                   -0.000025266043413665, -0.000124138519250119, -0.000151317604734284]

    # --- z_from_size ---
    # zc (j = 0,1,2; m = 0,1,2 → [z0,z1,z2])
    # FLAT: [z_00,z_01,z_02,  z_10,z_11,z_12,  z_20,z_21,z_22]
    z_from_size.zc: [-3.33078132514427, 0.0, 0.0,
                     4.54012609395602, 0.0, 0.0,
                     0.0, 0.0, 0.0]

    # limites de segurança para Z
    z_from_size.z_default: 0.0
    z_from_size.z_min: -10.0
    z_from_size.z_max: 10.0
